daprdocs/content/en/operations/monitoring/newrelic.md

```diff
@@ -0,0 +1,347 @@
+
+### State Store
+
+For state store I am using Redis and deploy it using the following command:
+
+```bash
+helm install redis bitnami/redis
+export REDIS_PASSWORD=$(kubectl get secret –namespace default \redis -o jsonpath=”{.data.redis-password}” | base64 –decode
+NAME            READY STATUS  RESTARTS  AGE
+redis-master-0  1/1   Running 0         16m
+redis-slave-0   1/1   Running 0         16m
+redis-slave-1   1/1   Running 0         16m
+```
+
+For further details, follow [these instructions](https://docs.dapr.io/getting-started/configure-redis/) to create and configure a Redis store
+
+## New Relic Instrumentation
+
+I have [forked](https://github.com/harrykimpel/quickstarts/tree/master/distributed-calculator) the above mentioned repo for the Distributed Calculator and updated the configuration to use New Relic code instrumentation and OpenTelemetry exporters using specific language agents or SDKs.
+
+In order for the integrations to send data to New Relic Telemetry Data Platform, you either need a [New Relic license key](https://docs.newrelic.com/docs/accounts/accounts-billing/account-setup/new-relic-license-key) or [New Relic Insights Insert API key](https://docs.newrelic.com/docs/apis/get-started/intro-apis/types-new-relic-api-keys#insights-insert-key). Both of these can easily be gathered/created from within the account you created in prerequisites.
+
+### New Relic Language agent
+
+As an example, the [New Relic agent instrumentation for .NET Core](https://docs.newrelic.com/docs/agents/net-agent/installation/install-docker-container) is part of the [Dockerfile](https://github.com/harrykimpel/quickstarts/blob/master/distributed-calculator/csharp/Dockerfile):
+
+```bash
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build-env
+WORKDIR /app
+
+# Copy csproj and restore as distinct layers
+COPY *.csproj ./
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+RUN dotnet publish -c Release -o out
+
+# Build runtime image
+FROM mcr.microsoft.com/dotnet/core/aspnet:3.1
+
+# Install the agent
+RUN apt-get update && apt-get install -y wget ca-certificates gnupg \
+    && echo 'deb http://apt.newrelic.com/debian/ newrelic non-free' | tee /etc/apt/sources.list.d/newrelic.list \
+    && wget https://download.newrelic.com/548C16BF.gpg \
+    && apt-key add 548C16BF.gpg \
+    && apt-get update \
+    && apt-get install -y newrelic-netcore20-agent
+
+# Enable the agent
+ENV CORECLR_ENABLE_PROFILING=1 \
+    CORECLR_PROFILER={36032161-FFC0-4B61-B559-F6C5D41BAE5A} \
+    CORECLR_NEWRELIC_HOME=/usr/local/newrelic-netcore20-agent \
+    CORECLR_PROFILER_PATH=/usr/local/newrelic-netcore20-agent/libNewRelicProfiler.so \
+    NEW_RELIC_LICENSE_KEY=<NEW-RELIC-LICENSE-KEY> \
+    NEW_RELIC_APP_NAME=distributed-calculator-dotnet-subtractor \
+    NEW_RELIC_DISTRIBUTED_TRACING_ENABLED=true
+
+WORKDIR /app
+COPY --from=build-env /app/out .
+ENTRYPOINT ["dotnet", "Subtract.dll"]
+```
+
+### OpenTelemetry instrumentation
```

Comment:
Based on the conversation with Mark and Nghia over email, is this needed here?

Reply:
@tcnghia can you please review this part specifically?

Reply:
I completely reworked the docs for this topic